GitUI/CommandsDialogs/RevisionDiff.cs

```diff
@@ -164,21 +155,56 @@ protected override void OnRuntimeLoad(EventArgs e)
 
         private string DescribeRevision(string sha1)
         {
+            return DescribeRevision(sha1, 0);
+        }
+
+        private string DescribeRevision(string sha1, int maxLength)
+        {
+            if (sha1.IsNullOrEmpty())
+            {
+                // No parent at all, present as working directory
+                return Strings.GetCurrentUnstagedChanges();
+            }
+
             var revision = _revisionGrid.GetRevision(sha1);
             if (revision == null)
             {
                 return sha1.ShortenTo(8);
             }
 
-            return _revisionGrid.DescribeRevision(revision);
+            return _revisionGrid.DescribeRevision(revision, maxLength);
+        }
+
+        /// <summary>
+        /// Provide a description for the first selected or parent to the "primary" selected last
+        /// </summary>
+        /// <returns>A description of the selected parent</returns>
+        private string DescribeSelectedParentRevision(bool showUnstagedAndCombined)
+        {
+            var parents = DiffFiles.SelectedItemParents
+                .Where(i => showUnstagedAndCombined ||
+                    !(i.Guid.IsNullOrWhiteSpace() || i.Guid == GitRevision.UnstagedGuid || i.Guid == DiffFiles.CombinedDiff.Text))
```

Comment:
what does `i.Guid == DiffFiles.CombinedDiff.Text` mean?
`CombinedDiff.Text` is a localised string "Combined Diff", why are we comparing it to a commit's SHA1?

Reply:
DiffFiles.CombinedDiff.Text is used both for display and as an ID for a variant of an "artificial commit".

Reply:
Something doesn't feel right about it.... can't articulate what exactly, but it feels bad

Reply:
It is not nice, but I preferred to not spread the knowledge about the revision. The proper way would be to add it in GitRevision.cs like:
        public const string IndexGuid = "1111111111111111111111111111111111111111";

Reply:
we already have it
https://github.com/gitextensions/gitextensions/blob/663036a7fe10d928596580e0c849107b8d00ddf3/GitCommands/Git/GitRevision.cs#L18

Reply:
In a similar way:
public const string CombinedDiffGuid = "2222...";
DiffFiles.CombinedDiff.Text will be used in some situations still.

Reply:
Added a CombinedDiffGuid for the artificial diff after all. For instance #4154 seem to enhance the comparison more, so it is better to set the pattern. Not sure if the TranslationApp handles this correctly.